src/main/java/com/puppycrawl/tools/checkstyle/checks/design/VisibilityModifierCheck.java

```diff
@@ -661,8 +659,7 @@ private boolean hasIgnoreAnnotation(DetailAST variableDef) {
      */
     private void visitImport(DetailAST importAst) {
         if (!isStarImport(importAst)) {
-            final DetailAST type = importAst.getFirstChild();
-            final String canonicalName = getCanonicalName(type);
+            final String canonicalName = getCanonicalName(importAst);
```

Comment:
This looks like another one connected to https://github.com/checkstyle/checkstyle/pull/12643#discussion_r1070767748 . The first child is always a period while the main ast is `IMPORT` token.